### ISSUE TITLE
Support case-insensitive match

### ISSUE THREAD

**Issue by th0ger** [garyelephant/pygrok#40]
I would be great if you could support case-insensitive matching.

Or elaborate the docs, if this is already possible?

**Comment by th0ger** [comment]
Both [regex](https://bitbucket.org/mrabarnett/mrab-regex/src/hg/) and [re](https://docs.python.org/3/library/re.html#re.search) support IGNORECASE arguments.

Maybe it would be possible to just pass *args and **kwargs onto regex/re?

**Comment by goldstar611** [comment]
I think winpat's PR, https://github.com/garyelephant/pygrok/pull/39 would give you what you need, right?

**Comment by th0ger** [comment]
Yes, I believe it would... whenever #39 build succeeds and a new pygrok version is released.

```
# python=3.8.5, pygrok=1.0.0

from pygrok import Grok
import re
text = 'gary IS male'
pattern = '%{WORD:name} is %{WORD:gender}'
grok = Grok(pattern, flags=re.IGNORECASE)
print (grok.match(text))

# -> unexpected keyword argument 'flags'
```
